web/src/model/topology.ts

```diff
@@ -399,23 +399,31 @@ export const generateDataModel = (
       nodes.push(group);
     }
 
-    if (parent && !parent.children!.includes(group.id)) {
-      parent.children!.push(group.id);
+    if (parent) {
+      if (parent.id == group.id) {
+        console.error('addGroup parent === group', parent, group);
+      } else if (!parent.children!.includes(group.id)) {
+        parent.children!.push(group.id);
+      }
     }
 
     return group;
   };
 
   const addNode = (data: NodeData, scope: MetricScope): NodeModel => {
-    const parent = addPossibleGroups(data.peer);
-    let node = nodes.find(n => n.id === data.peer.id);
+    const parent = data.nodeType !== 'unknown' ? addPossibleGroups(data.peer) : undefined;
+    let node = nodes.find(n => n.type === 'node' && n.id === data.peer.id);
```

Comment:
The root cause of this issue is that `unknown` nammed items were generating nodes with the same peer id as groups.

To avoid such issues 'unknown' nodes can't have parents + nodes find function will return only `node` types